app/main.py

```diff
@@ -1,3 +1,48 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other):
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, num):
+        self.km *= num
+        return self
```

Comment:
in this method you modify `self.km`, but you shouldn't do this. You have to return a different `Distance` object and pass to it a new value, that is calculated by `self.km * num`